From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mechoriet <kevinworm92@gmail.com>
Date: Fri, 11 Nov 2022 23:08:10 +0100
Subject: [PATCH] Fix AIOOBE in inventory handling


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index d812ae47c7e3b2bfb2c5af4ef30b5d4a015fff3b..83bbd7cfdd4125bbf509283fd4840a9c8bd077e1 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -1631,7 +1631,7 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                 } else if (packetplayinwindowclick.f() == 3) {
                     if (packetplayinwindowclick.c() == 2) {
                         click = ClickType.MIDDLE;
-                        if (packetplayinwindowclick.b() == -999) {
+                        if (packetplayinwindowclick.b() < 0) { // PandaSpigot - Paper Issue GH-404
                             action = InventoryAction.NOTHING;
                         } else {
                             Slot slot = this.player.activeContainer.getSlot(packetplayinwindowclick.b());
