From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: mechoriet <kevinworm92@gmail.com>
Date: Fri, 18 Nov 2022 02:04:58 +0100
Subject: [PATCH] Cache block break animation packet


diff --git a/src/main/java/net/minecraft/server/WorldManager.java b/src/main/java/net/minecraft/server/WorldManager.java
index 5c06ca24ba4c6fea74bd73b5e43b598f78b35139..7b4ca05df836d44f88e3569023d04732509efc96 100644
--- a/src/main/java/net/minecraft/server/WorldManager.java
+++ b/src/main/java/net/minecraft/server/WorldManager.java
@@ -66,6 +66,7 @@ public class WorldManager implements IWorldAccess {
         if (entity instanceof EntityHuman) entityhuman = (EntityHuman) entity;
         // CraftBukkit end
 
+        PacketPlayOutBlockBreakAnimation packet = null; // PandaSpigot - Cache block break animation packet
         while (iterator.hasNext()) {
             EntityPlayer entityplayer = (EntityPlayer) iterator.next();
 
@@ -81,7 +82,10 @@ public class WorldManager implements IWorldAccess {
                 // CraftBukkit end
 
                 if (d0 * d0 + d1 * d1 + d2 * d2 < 1024.0D) {
-                    entityplayer.playerConnection.sendPacket(new PacketPlayOutBlockBreakAnimation(i, blockposition, j));
+                    // PandaSpigot start - Cache block break animation packet
+                    if (packet == null) packet = new PacketPlayOutBlockBreakAnimation(i, blockposition, j);
+                    entityplayer.playerConnection.sendPacket(packet);
+                    // PandaSpigot end
                 }
             }
         }
