From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mechoriet <kevinworm92@gmail.com>
Date: Fri, 6 Oct 2023 21:18:05 +0200
Subject: [PATCH] Implement EntityPickupItemEvent


diff --git a/src/main/java/net/minecraft/server/EntityArrow.java b/src/main/java/net/minecraft/server/EntityArrow.java
index f9f0c0ac9196c581d1eeb3c40dce93fee913d13a..02c905051e8bfb73cbc166f58d1f4d313ed9c88a 100644
--- a/src/main/java/net/minecraft/server/EntityArrow.java
+++ b/src/main/java/net/minecraft/server/EntityArrow.java
@@ -443,6 +443,16 @@ public class EntityArrow extends Entity implements IProjectile {
                 if (event.isCancelled()) {
                     return;
                 }
+                // PandaSpigot start - EntityPickupItemEvent
+                // Call newer event afterwards
+                org.bukkit.event.entity.EntityPickupItemEvent entityEvent = new org.bukkit.event.entity.EntityPickupItemEvent(entityhuman.getBukkitEntity(), event.getItem(), 0);
+                // event.setCancelled(!entityhuman.canPickUpLoot); TODO
+                this.world.getServer().getPluginManager().callEvent(entityEvent);
+
+                if (entityEvent.isCancelled()) {
+                    return;
+                }
+                // PandaSpigot end
             }
             // CraftBukkit end
             boolean flag = this.fromPlayer == 1 || this.fromPlayer == 2 && entityhuman.abilities.canInstantlyBuild;
diff --git a/src/main/java/net/minecraft/server/EntityInsentient.java b/src/main/java/net/minecraft/server/EntityInsentient.java
index cd3079bae2604ed8d03ce0e3bcc582f354248940..0c6a2912e4ec144abf5d67a807b232888a15a746 100644
--- a/src/main/java/net/minecraft/server/EntityInsentient.java
+++ b/src/main/java/net/minecraft/server/EntityInsentient.java
@@ -426,7 +426,14 @@ public abstract class EntityInsentient extends EntityLiving {
                 }
             }
 
-            if (flag && this.a(itemstack)) {
+            // PandaSpigot start - EntityPickupItemEvent
+            boolean canPickup = flag && this.a(itemstack);
+            org.bukkit.event.entity.EntityPickupItemEvent entityEvent = new org.bukkit.event.entity.EntityPickupItemEvent((org.bukkit.entity.LivingEntity) this.getBukkitEntity(), (org.bukkit.entity.Item) entityitem.getBukkitEntity(), 0);
+            entityEvent.setCancelled(!canPickup);
+            this.world.getServer().getPluginManager().callEvent(entityEvent);
+
+            if (!entityEvent.isCancelled()) {
+            // PandaSpigot end
                 if (itemstack1 != null && this.random.nextFloat() - 0.1F < this.dropChances[i]) {
                     this.a(itemstack1, 0.0F);
                 }
diff --git a/src/main/java/net/minecraft/server/EntityItem.java b/src/main/java/net/minecraft/server/EntityItem.java
index b59ffc5ea31bfbebef2f45bf8cdaf9c166e0834a..b06d4dda99d72511e3838cd025d3d4d61b46a7a8 100644
--- a/src/main/java/net/minecraft/server/EntityItem.java
+++ b/src/main/java/net/minecraft/server/EntityItem.java
@@ -314,11 +314,21 @@ public class EntityItem extends Entity {
                 PlayerPickupItemEvent event = new PlayerPickupItemEvent((org.bukkit.entity.Player) entityhuman.getBukkitEntity(), (org.bukkit.entity.Item) this.getBukkitEntity(), remaining);
                 // event.setCancelled(!entityhuman.canPickUpLoot); TODO
                 this.world.getServer().getPluginManager().callEvent(event);
-                itemstack.count = canHold + remaining;
+                //itemstack.count = canHold + remaining; // PandaSpigot - move down after new event
 
                 if (event.isCancelled()) {
                     return;
                 }
+                // PandaSpigot start - EntityPickupItemEvent
+                // Call newer event afterwards
+                org.bukkit.event.entity.EntityPickupItemEvent entityEvent = new org.bukkit.event.entity.EntityPickupItemEvent(entityhuman.getBukkitEntity(), (org.bukkit.entity.Item) this.getBukkitEntity(), remaining);
+                // event.setCancelled(!entityhuman.canPickUpLoot); TODO
+                this.world.getServer().getPluginManager().callEvent(entityEvent);
+                if (entityEvent.isCancelled()) {
+                    return;
+                }
+                itemstack.count = canHold + remaining;
+                // PandaSpigot end
 
                 // Possibly < 0; fix here so we do not have to modify code below
                 this.pickupDelay = 0;
