From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: CrimsonEdgeHope <92579614+crimsonedgehope@users.noreply.github.com>
Date: Thu, 20 Feb 2025 17:36:23 +0800
Subject: [PATCH] Publish


diff --git a/build.gradle.kts b/build.gradle.kts
index 685816802b45940613f7f1e8199687efa520fb63..decd0753cf631f950b9c908d44821c7ec7fe4ceb 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -1,6 +1,9 @@
+import java.net.URI
+
 plugins {
     id("pandaspigot.conventions")
     id("com.github.johnrengelman.shadow") version "7.1.2"
+    `maven-publish`
 }
 
 val minecraftVersion = "1_8_R3"
@@ -153,3 +156,34 @@ tasks {
         dependsOn(named("remap"))
     }
 }
+
+publishing {
+    publications {
+        create<MavenPublication>("mavenJava") {
+            artifact(tasks.named("remap"))
+            versionMapping {
+                usage("java-api") {
+                    fromResolutionOf("runtimeClasspath")
+                }
+            }
+        }
+    }
+
+    (System.getenv("CUSTOMSPIGOT_MAVEN_URL")).let { urlString ->
+        if (urlString == null) {
+            return@let
+        }
+
+        repositories {
+            maven {
+                name = "CustomSpigot"
+                url = URI.create(urlString)
+
+                credentials {
+                    username = System.getenv("CUSTOMSPIGOT_MAVEN_USERNAME")
+                    password = System.getenv("CUSTOMSPIGOT_MAVEN_PASSWORD")
+                }
+            }
+        }
+    }
+}
